proposed/vcpkg.md

```diff
@@ -0,0 +1,118 @@
+# Using vcpkg in .NET projects and the VMR
+
+**Owner** [Jeremy Koritzinsky](https://github.com/jkoritzinsky)
+
+## Scenarios and User Experience
+
+In .NET today, we handle native dependencies and native dependency flow by a few different mechanisms:
+
+- Vendoring dependencies, aka including a copy of the source with our applied patches in the product repository.
+- Git submodules pointing to the source repo of the dependency.
+- NuGet packages that flow pre-built between our repositories.
+
+For many of our dependencies, these build options work decently well. However, these mechanisms have a few drawbacks:
+
+- Vendoring code increases the size of our repositories and increases build times.
+- Submodules aren't cloned by default and have a worse UX than vendoring code, and cannot support patches.
+- NuGet packages are not easy to consume from native CMake projects, so we end up having to wire up MSBuild to consume them and construct paths to pass to CMake.
+
+Additionally, [the VMR design documents on external source](https://github.com/dotnet/arcade/blob/main/Documentation/UnifiedBuild/VMR-Strategy-For-External-Source.md) defines that we shouldn't use submodules directly in our build.
+
+For native dependencies, we could alternatively use Vcpkg, a cross-platform C++ package manager provided by another team at Microsoft.
+Vcpkg would make it easier for us to both consume native dependencies in our projects, as many C++ libraries that we may consider using are already available in vcpkg.
+It would also make it easier for us to contribute our own native projects, like the nethost library, back to vcpkg in an official manner.
+
+## Requirements
+
+- We must be able to version and patch dependencies ourselves.
+- We must be able to produce private patches that we can consume without public publishing.
+- We must be able to consume dependencies from vcpkg in our projects.
+- It must be possible to use vcpkg in offline build scenarios.
+- It must be possible to use vcpkg in the VMR builds.
+
+### Goals
+
+- Enable developers of .NET to consume native dependencies from vcpkg.
```

Comment:
I do not think that "use tech XYZ" should be the goal for eng infrastructure investments.

The goal for infrastructure investment should be to make the eng infrastructure cheaper, simpler, faster, etc. For example, check   https://github.com/dotnet/arcade/blob/05493e05d4bbf262f1be1bd517ac95f5bff3a2ef/Documentation/UnifiedBuild/Overview.md#goals .

Reply:
I'll rephrase this point. My main idea here was to show how this engineering improvement could assist us in more easily shipping an official nethost vcpkg package (something our customers want and some members of our team unofficially maintain) and provide some product benefit in addition to the infrastructure benefits.

Reply:
> could assist us in more easily shipping an official nethost vcpkg package

What makes vcpkg packages official? My understanding is that the vcpkg packages are published by merging a PR to https://github.com/microsoft/vcpkg. Would that step go away somehow?

(In any case, we can reduce the patch set required to publish the package, but that should not require introducing vcpkg official build dependency.)

Reply:
I'm not sure exactly what makes the packages official, but I know that they explicitly mark unofficial packages as unofficial by putting targets generated by them in the `unofficial` namespace to ensure they won't clash with official packages. I agree that we can make official vcpkg packages without using vcpkg ourselves, but using vcpkg ourselves makes things a little easier.

Reply:
I took a closer look at what makes a package "official". The "official" vs "unofficial" terminology mainly refers to which CMake targets are available. If a target is exported from CMake by a patch added in the portfile, then it is an "unofficial" target and should be namespaced as such. If the target is exported by the CMake scripts in the source repository, then the targets are considered "official".

I'll remove this goal as it looks like we would just need to upstream the patches to install CMake exports in our build scripts to make them "official"